packages/core/src/version.ts

```diff
@@ -7,36 +7,46 @@ export interface Version {
   linkedWithoutMetadata: string;
 }
 
-export function getVersion(bytecode: string, linkedBytecode?: string): Version {
+export function getVersion(bytecode: string, linkedBytecode?: string, constructorArgs?: string): Version {
   if (bytecode !== '') {
     return {
       withMetadata: hashBytecode(bytecode),
       withoutMetadata: hashBytecodeWithoutMetadata(bytecode),
-      linkedWithoutMetadata: hashBytecodeWithoutMetadata(linkedBytecode ?? bytecode),
+      linkedWithoutMetadata: hashBytecodeWithoutMetadata(linkedBytecode ?? bytecode, constructorArgs),
     };
   } else {
     throw new Error('Abstract contract not allowed here');
   }
 }
 
-export function hashBytecode(bytecode: string): string {
+export function hashBytecode(bytecode: string, constructorArgs?: string): string {
   bytecode = bytecode
     .replace(/__\$([0-9a-fA-F]{34})\$__/g, (_, placeholder) => `000${placeholder}000`)
     .replace(/__\w{36}__/g, placeholder => keccak256(Buffer.from(placeholder)).toString('hex', 0, 20));
 
+  constructorArgs ??= '';
+
   // Fail if bytecode is still not valid hex after transformations above.
   // Buffer.from silently truncates at the first non-hex character.
   // NOTE: Some bytecode seems to have odd length, so we cannot do ([0-9a-fA-F]{2})*
   if (!/^(0x)?[0-9a-fA-F]*$/.test(bytecode)) {
     throw new Error('Bytecode is not a valid hex string');
   }
 
-  const buf = Buffer.from(bytecode.replace(/^0x/, ''), 'hex');
+  if (!/^(0x)?[0-9a-fA-F]*$/.test(constructorArgs)) {
+    throw new Error('Constructor arguments are not a valid hex string');
+  }
+
+  const buf = Buffer.concat([
```

Comment:
Ok on second thought I don't think we should be concerned. The concatenation of bytecode + args is exactly what is sent with the deploy transaction. If the result of concatenating (B1, A1) and (B2, A2) is the same C creation code, then it seems we can safely assume it will behave exactly the same.